src/command/restore/restore.c

```diff
@@ -247,52 +327,80 @@ restoreBackupSet(InfoBackup *infoBackup)
                         // If the end of the backup is before the target time, then select this backup
                         if (backupData.backupTimestampStop < timeTargetEpoch)
                         {
-                            backupSet = backupData.backupLabel;
+                            found = true;
+
+                            result = restoreBackupData(
+                                backupData.backupLabel, repoIdx, infoPgCipherPass(infoBackupPg(infoBackup)));
                             break;
                         }
                     }
 
-                    if (backupSet == NULL)
+                    // If a backup was found on this repo matching the criteria for time then exit, else determine if the latest
+                    // backup from this repo might be used
+                    if (found)
+                        break;
+                    else
                     {
-                        LOG_WARN_FMT(
-                            "unable to find backup set with stop time less than '%s', latest backup set will be used",
-                            strZ(cfgOptionStr(cfgOptTarget)));
-                    }
-                }
-            }
+                        InfoBackupData latestBackup = infoBackupData(infoBackup, infoBackupDataTotal(infoBackup) - 1);
```

Comment:
We get this data twice, here and line 356. Perhaps we should just move this up to 316? I'm not worried about the performance hit but de-duping the code might be nice.

Reply:
Fixed f20b7b7d8031a0615a645f57aec223454c7fbc19